src/tc/actions/action.rs

```diff
@@ -165,7 +203,17 @@ impl Nla for TcActionAttribute {
     fn kind(&self) -> u16 {
         match self {
             Self::Kind(_) => TCA_ACT_KIND,
-            Self::Options(_) => TCA_ACT_OPTIONS,
+            Self::Options(opts) => {
+                // NOTE: the kernel simply doesn't consistently use the nested
+                // flag based on captured messages.
+                // This is heuristically trying to match the kernel's behavior
+                // but may not be correct.
+                if opts.len() == 1 {
+                    TCA_ACT_OPTIONS | NLA_F_NESTED
```

Comment:
In kernel we have 

```c
static const struct nla_policy tcf_action_policy[TCA_ACT_MAX + 1] = {
        [TCA_ACT_KIND]          = { .type = NLA_STRING },
        [TCA_ACT_INDEX]         = { .type = NLA_U32 },
        [TCA_ACT_COOKIE]        = { .type = NLA_BINARY,
                                    .len = TC_COOKIE_MAX_SIZE },
        [TCA_ACT_OPTIONS]       = { .type = NLA_NESTED },
```

And `nest = nla_nest_start_noflag(skb, TCA_ACT_OPTIONS);`.

So it should always carry with `NLA_F_NESTED`.

Reply:
Ah, it must be iproute2 just being wrong sometimes.  I will change our code to always set NLA_F_NESTED